src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -126,21 +126,36 @@ public override
         {
             get
             {
-                if (ClassImpl == typeof(decimal) || ClassImpl == typeof(decimal?))
+                // In case of multiple CustomAttributes the following rules apply:
+                // - if declared type is DateTime, then DateTimeConstantAttribute is favoured over others
+                // - else if there is at least one CustomConstantAttribute, then it is favoured over others
+                // - else use the first attribute providing the default value
+                if (ClassImpl == typeof(DateTime) || ClassImpl == typeof(DateTime?))
```

Comment:
This code does not behave exactly same as CoreCLR. For example, this prints nothing with CoreCLR but it prints a date with Mono:
```
using System.Diagnostics;
using System.Reflection;
using System.Runtime.CompilerServices;

public class Program
{
    static void Main()
    {
        Console.WriteLine(typeof(Program).GetMethod("M").GetParameters()[0].RawDefaultValue);
    }

    public static void M([DateTimeConstant(42_000_000)][AnotherDateTimeCustomConstant] DateTime? d)
    {
    }
}

class AnotherDateTimeCustomConstantAttribute : CustomConstantAttribute
{
    public override object Value => new DateTime(430_000_000);
}
```

Reply:
Thank you very much for drawing my attention to this.
I am aware of the issue and did not want to move this PR from draft because of it.
I have however opened a new tracking [issue](https://github.com/dotnet/runtime/issues/76174) to reflect the RawDefaultValue resolution, since it is a bit broader then the original one.
I will continue working on this.

Reply:
Sounds good. Keep in mind my other comment that it is ok to adjust CoreCLR behavior. The current code in this PR that tries to match the current CoreCLR behavior at all costs does not feel like an improvement.

What would be the simplest ordering rules? I would start with answering that, and then fix everything to match them.